service/src/main/java/uk/gov/hmcts/reform/fpl/service/HearingVenueLookUpService.java

```diff
@@ -0,0 +1,46 @@
+package uk.gov.hmcts.reform.fpl.service;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.fpl.model.HearingVenue;
+import uk.gov.hmcts.reform.fpl.utils.ResourceReader;
+
+import java.io.IOException;
+import java.util.List;
+
+@Slf4j
+@Service
+public class HearingVenueLookUpService {
+
+    private final ObjectMapper objectMapper;
+
+    @Autowired
+    public HearingVenueLookUpService(ObjectMapper objectMapper) {
+        this.objectMapper = objectMapper;
+    }
+
+    private List<HearingVenue> getHearingVenueMappings() throws IOException {
+        final String jsonContent = ResourceReader.readString("static_data/hearingVenues.json");
+        return objectMapper.reader()
+            .forType(new TypeReference<List<HearingVenue>>() {})
+            .readValue(jsonContent);
+    }
+
+    HearingVenue getHearingVenue(final String venueId) throws IOException {
+        List<HearingVenue> hearingVenues = getHearingVenueMappings();
+
+        return hearingVenues.stream()
+            .filter(hearingVenue -> venueId.equalsIgnoreCase(hearingVenue.getHearingVenueId()))
+            .findFirst()
+            .orElse(HearingVenue.builder().build());
+    }
+
+    String buildHearingVenue(final HearingVenue hearingVenue) {
+        return String.join(", ", hearingVenue.getAddress().getAddressLine1(),
+            hearingVenue.getAddress().getAddressLine2(), hearingVenue.getAddress().getPostTown(),
```

Comment:
Do we have those values in 100% scenarios?

Reply:
for this particular case, we do. But updated with extra checks just in case the [hearVenues.json](https://github.com/hmcts/fpl-ccd-configuration/blob/7675b8d0a31a8bad26b930fb129c4652970e7877/service/src/main/resources/static_data/hearingVenues.json) is updated down the line. thanks for spotting.